backend/app/Actions/Visitors/BounceRateAction.php

```diff
@@ -32,8 +32,11 @@ public function execute(BounceRateRequest $request): BounceRateResponse
         $collection = new Collection();
         do {
             $all = $allVisitorsByTimeFrameValues[$start]??0;
+            if ($all === 0) {
```

Comment:
filtering of empty values is not good approach, as it will show a falsy result. Better trimming zero result from both sides start and end